### DISCUSSION TITLE
Deprecation of EC4/EC63/EC75 rule for JAVA

### DISCUSSION THREAD

**Discussion by alograg** [green-code-initiative/ecoCode#271] [Q&A]
Hi @dedece35 ,

I'm writing to inquire about the deprecation of the EC4/EC63/EC75 rule for java (issue #228, #233, #235) . As you know, this rule has been marked as deprecated.

I'm wondering if you have a timeline for removing this rule from the codebase. I'd also like to know when the next release of the project on GitHub is planned. The current release is version 1.4.4, I know that the tags exist until version 1.4.7.

Thank you for your time and attention to this matter.

Sincerely,

**Comment by dedece35** [comment]
Hi @alograg,

our internal rule to remove deprecated rules is that they will be removed in one of two next minor versions (X+1 or X+2) (where in X.Y.Z, X is major, Y is minor and Z is patch).
We have just released our plugins. You can follow changelog (or releases page) here : 
- old java / python / php plugins : 
   - https://github.com/green-code-initiative/ecoCode/blob/main/CHANGELOG.md
   - https://github.com/green-code-initiative/ecoCode/releases
- new java plugin : 
   - https://github.com/green-code-initiative/ecoCode-java/blob/main/CHANGELOG.md
   - https://github.com/green-code-initiative/ecoCode-java/releases
- new PHP plugin : 
   - https://github.com/green-code-initiative/ecoCode-php/blob/main/CHANGELOG.md
   - https://github.com/green-code-initiative/ecoCode-php/releases
- new Python plugin : 
   - https://github.com/green-code-initiative/ecoCode-python/blob/main/CHANGELOG.md
   - https://github.com/green-code-initiative/ecoCode-python/releases

thus, next java plugin 1.6.0, next python plugin 1.5.0 and next php plugin 1.5.0 will remove deprecated rules.

regards,
David.

**Reply by alograg** [reply]
Merci
